<launch>
    <arg name="rviz_args" value="-d $(find people_tracking)/extra/multipeople_tracking.rviz" />
    <arg name="map_file" default="$(find people_tracking)/extra/etu_map.yaml"/>
    <arg name="scan_topic" value="scan"/>
    <arg name="scan_frame_id" value="laser"/>

    <!-- Core tracking nodes-->
    <node pkg="people_tracking" name="wall_filter" type="wall_filter.py">
        <param name="scan_topic" value="$(arg scan_topic)"/>
    </node>
    <node pkg="people_tracking" name="multiperson_tracker" type="multiperson_tracker.py" output="screen"/>
    <node pkg="people_tracking" name="reset" type="reset_walls.py" output="screen"/>

    <!-- Place markers for people's locations -->
   <node pkg="people_tracking" name="people_marker" type="mark_people.py" output="screen">
        <param name="scale_factor" value="2.0"/>
        <param name="person_height" value="0.5"/>
        <param name="multiperson" value="true"/>
    </node> 
  
    <!-- Load map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />
    
    <!-- transform laser to frame of map-->
    <node pkg="tf" type="static_transform_publisher" name="laser_to_map" args="0.3 1.65 0 0 0 0 1 /map /$(arg scan_frame_id) 100"/>

    <!-- Visualize in rviz -->
    <node name="$(anon rviz)" pkg="rviz" type="rviz" required="false" 
        respawn="false" output="screen" args="$(arg rviz_args)">
    </node>
</launch>
